src/NLog/Targets/Wrappers/LimitingTargetWrapper.cs

```diff
@@ -0,0 +1,174 @@
+//   contributors may be used to endorse or promote products derived from this
+//   software without specific prior written permission. 
+// 
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE 
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE 
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE 
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR 
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN 
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) 
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF 
+// THE POSSIBILITY OF SUCH DAMAGE.
+// 
+
+namespace NLog.Targets.Wrappers
+{
+    using System;
+    using System.ComponentModel;
+    using NLog.Common;
+    using NLog.Time;
+
+
+    /// <summary>
+    /// Limits the number of messages written per timespan to the wrapped target.
+    /// </summary>
+    [Target("LimitingWrapper", IsWrapper = true)]
+    public class LimitingTargetWrapper : WrapperTargetBase
+    {
+        private DateTime firstWriteInInterval;
+        private int writeCount;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LimitingTargetWrapper" /> class.
+        /// </summary>
+        public LimitingTargetWrapper():this(null)
+        {
+        }
+
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LimitingTargetWrapper" /> class.
+        /// </summary>
+        /// <param name="name">The name of the target.</param>
+        /// <param name="wrappedTarget">The wrapped target.</param>
+        public LimitingTargetWrapper(string name, Target wrappedTarget) 
+            : this(wrappedTarget, 1000, TimeSpan.FromHours(1))
+        {
+            this.Name = name;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LimitingTargetWrapper" /> class.
+        /// </summary>
+        /// <param name="wrappedTarget">The wrapped target.</param>
+        public LimitingTargetWrapper(Target wrappedTarget)
+            : this(wrappedTarget, 1000, TimeSpan.FromHours(1))
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LimitingTargetWrapper" /> class.
+        /// </summary>
+        /// <param name="wrappedTarget">The wrapped target.</param>
+        /// <param name="messageLimit">Maximum number of messages written per interval.</param>
+        /// <param name="interval">Interval in which the maximum number of messages can be written.</param>
+        public LimitingTargetWrapper(Target wrappedTarget, int messageLimit, TimeSpan interval)
```

Comment:
Mh, I provided the same overloads as the `AsyncTargetWrapper `and `BufferingTargetWrapper`. I think those are good. Do you really think more would add value?